api/src/test/java/org/apache/iceberg/transforms/TestResiduals.java

```diff
@@ -191,26 +191,18 @@ public void testInTimestamp() {
         Types.NestedField.optional(51, "dateint", Types.IntegerType.get())
     );
 
-    Long date20191201 = (Long) Literal.of("2019-12-01T00:00:00.00000")
-        .to(Types.TimestampType.withoutZone()).value();
-    Long date20191202 = (Long) Literal.of("2019-12-02T00:00:00.00000")
-        .to(Types.TimestampType.withoutZone()).value();
-
     PartitionSpec spec = PartitionSpec.builderFor(schema)
         .day("ts")
         .build();
 
-    Transform day = spec.getFieldsBySourceId(50).get(0).transform();
-    Integer tsDay = (Integer) day.apply(date20191201);
-
-    Predicate pred = in("ts", date20191201, date20191202);
+    Predicate pred = in("ts", "2019-12-01T00:00:00.00000", "2019-12-01T00:00:00.00000");
```

Comment:
Why were these changes necessary?

Reply:
The unit tests failed with the following exception, the same reason that tranform.day doesn't produce the date value. These changes are not necessary just use to pass unit tests. I will revert the changes.
```
Cannot cast java.lang.Integer to java.time.LocalDate
java.lang.ClassCastException: Cannot cast java.lang.Integer to java.time.LocalDate
	at java.lang.Class.cast(Class.java:3369)
	at org.apache.iceberg.TestHelpers$Row.get(TestHelpers.java:121)
	at org.apache.iceberg.Accessors$DateAccessor.get(Accessors.java:216)
	at org.apache.iceberg.Accessors$DateAccessor.get(Accessors.java:209)
	at org.apache.iceberg.expressions.BoundReference.eval(BoundReference.java:39)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.in(ResidualEvaluator.java:201)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.in(ResidualEvaluator.java:120)
	at org.apache.iceberg.expressions.ExpressionVisitors$BoundExpressionVisitor.predicate(ExpressionVisitors.java:152)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.predicate(ResidualEvaluator.java:258)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.predicate(ResidualEvaluator.java:281)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.predicate(ResidualEvaluator.java:120)
	at org.apache.iceberg.expressions.ExpressionVisitors.visit(ExpressionVisitors.java:283)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.eval(ResidualEvaluator.java:125)
	at org.apache.iceberg.expressions.ResidualEvaluator$ResidualVisitor.access$100(ResidualEvaluator.java:120)
	at org.apache.iceberg.expressions.ResidualEvaluator.residualFor(ResidualEvaluator.java:117)
	at org.apache.iceberg.transforms.TestResiduals.testInTimestamp(TestResiduals.java:211)
```